src/contracts/Harvester.sol

```diff
@@ -0,0 +1,267 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import {Initializable} from "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import {IERC20, IERC20Metadata} from "@openzeppelin/contracts/token/ERC20/extensions/IERC20Metadata.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {Math} from "@openzeppelin/contracts/utils/math/Math.sol";
+
+import {OwnableOperable} from "./OwnableOperable.sol";
+import {IHarvestable, IMagpieRouter, IOracle} from "./Interfaces.sol";
+
+/**
+ * @title Collects rewards from strategies and swaps them for ARM liquidity tokens.
+ * @author Origin Protocol Inc
+ */
+contract Harvester is Initializable, OwnableOperable {
```

Comment:
Maybe we should give this contract a slightly different name, since we already have a quite different "Harvester" contract already.

Reply:
Good point. I'll rename to `SonicHarvester` as it has Sonic specific logic. eg Magpie DEX aggregator